### DISCUSSION TITLE
Berechnung E-Auto Anteil optimieren

### DISCUSSION THREAD

**Discussion by Bart1909** [solectrus/solectrus#2854] [General]
Hi Georg,

wir sind mittlerweile auch Besitzer eines E-Autos und jetzt, wo die Sonne langsam mal etwas mehr raus kommt, ist mir aufgefallen, dass die Berechnung des Anteils vom E-Auto bei der Preisübersicht nicht ganz optimal ist.

Wenn ich es richtig sehe, wird bei der Tagesansischt geschaut, wie hoch der Netzbezug war (in meinem Fall gestern 39,7kWh). Dann wird geschaut, wie viel ins Auto gegeangen ist (36,2kWh). Diese 36,2kWh werden dann komplett aufs E-Auto geschustert, unabhängig davon, ob während des Ladevorgangs Energie vom Dach ins Auto ging oder nicht.
![image](https://github.com/solectrus/solectrus/assets/14097923/c547aafa-04e6-4e84-855b-61a14a42c6d9)
![image](https://github.com/solectrus/solectrus/assets/14097923/2c751cd3-eb11-4c00-a050-c005494dccc1)

Stimmt das so? Falls ja, könnte man die Berechnung nicht anpassen? Es ist ja gespeichert, wann wie viel ins Auto ging und wie hoch zu diesem Zeitpunkt der Netzbezug war. Dann könnte man nur den Netzbezug zu dieser Zeit dem E-Auto zuweisen. Auch nicht 100% korrekt, aber sollte genauer sein.

Viele Grüße

Tobi

**Comment by StevieC121176** [comment]
Ja das ist mir auch schon aufgefallen. Da wir unsere Autos mit Evcc laden ist unser Jahresdurchschnitt bei knapp 90% Sonnenanteil fürs Autoladen, da fast nur mit Überschuss geladen wird.

**Comment by ledermann** [comment]
Ja, gut beobachtet, die "abgeschätzte Aufteilung" ist nur eine Heuristik, die grob daneben liegen kann.
Es gibt in diesem Zusammenhang übrigens auch #2436.

Insgesamt bin ich mit der jetzigen Berechnung auch nicht glücklich und würde das gerne genauer machen. Das würde bedeuten, nicht die Summen (pro Tag, Woche, Monat, ...) zu betrachten, sondern die Berechnung auf kleinste Zeiträume herunterzubrechen. Das würde dann wirklich die Frage beantworten, wo der Strom herkommt, der ins Auto fließt. 

Rein logisch ist klar, was gemacht werden muss. Die Herausforderung ist aber, die InfluxDB so abzufragen, dass es auch performant ist. Daran scheitert es bislang. Ich sehe mir das aber noch mal an.

**Reply by Bart1909** [reply]
Ah, #2436 hatte ich gar nicht gefunden, sorry.

Okay, vielleicht findest du ja einen guten Weg :)
